### PR TITLE
Fix for test_snapshots.py using nfs2 instead of nfs template

### PR DESCRIPTION
**Pull request by nvazquez**
Fix for marvin test failure introduced in #1847 

Cc: @borisstoyanov @rhtyd @karuturi

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangurtan package

**Comment by borisstoyanov** [comment]
Thanks for this fix @nvazquez, I'll kick marvin tests to verify it doesn't brake any other test.

**Comment by nvazquez** [comment]
@borisstoyanov great, thanks!

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by serg38** [comment]
LGTM

**Comment by karuturi** [comment]
@nvazquez Are you looking at the test failures?

**Comment by nvazquez** [comment]
@karuturi @borisstoyanov next step will be updating Marvin's folder `test_data.py` file and configure an url for key "nfs2" similar as it was done for "nfs", this way it can be used to mount nfs in running environment.

**Comment by serg38** [comment]
@karuturi I agree with @nvazquez . This PR can be safely merged and then B.O. runtime environment needs to be adjusted to define nfs2 mountpoint in test_data.py

**Comment by rhtyd** [comment]
@serg38 okay, we can add this to test_data /cc @PaulAngus

**Comment by nvazquez** [comment]
Hi @karuturi @rhtyd @borisstoyanov,
I've pushed a new commit for fixing `test_snapshots.py` failure. Can you please run tests against Vmware and Kvm?

These were results in our env, using Vmware:
````
[root@ussarlabcsmgt41 cloudstack]# cat /tmp//MarvinLogs//XJVTRH/results.txt
Test Snapshot Root Disk ... === TestName: test_01_snapshot_root_disk | Status : SUCCESS ===
ok
Test listing volume snapshots with removed data stores ... === TestName: test_02_list_snapshots_with_removed_data_store | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 2 tests in 410.231s

OK
````

**Comment by borisstoyanov** [comment]
thanks @nvazquez let me package again
@blueorangutan package

**Comment by borisstoyanov** [comment]
@nvazquez I'm getting the following error:
```
File "/marvin/tests/smoke/test_snapshots.py", line 327, in test_02_list_snapshots_with_removed_data_store\n    livemigrate="true"\n', '  
{errorcode : 431, errortext : u'Volume needs to be detached from VM'}, accountid : u'ded4cc41-ff1f-11e6-95d0-065ddc010723'}\n"]
```
does the test detach the VM before migrating it? didn't managed to find that part looking in the code.

File: test/integration/smoke/test_snapshots.py [hunk 1]
```diff
@@ -275,7 +275,7 @@ def test_02_list_snapshots_with_removed_data_store(self):
         assert isinstance(clusters,list) and len(clusters)>0
 
         storage = StoragePool.create(self.apiclient,
```
Comment (nvazquez) [inline test/integration/smoke/test_snapshots.py]: Done, thanks for pointing it out!

**Comment by nvazquez** [comment]
@borisstoyanov actually disk is not being dettached from vm before migrating it, it is using vm's ROOT disk, it can be done on Vmware by setting `livemigrate='true'` to migrate command. I'll work on refactoring the test so that migration can be achieved

**Comment by nvazquez** [comment]
@borisstoyanov I refactored marvin test to migrate a detached disk instead of vm's root volume as it was before. Can you please test it again in your env?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by serg38** [comment]
@rhtyd @borisstoyanov did you have a chance to add nfs2 test data in B.O. ?

**Comment by borisstoyanov** [comment]
Yes @serg38, but currently testing it. So this batch doesn't have it.

**Comment by borisstoyanov** [comment]
@nvazquez @serg38 I'm happy to merge this change I've run the tests with the NFS2 trillian branch and it passed. As soon as we get the trillian branch merged as well we'll have this passing in the smoketests.

**Comment by nvazquez** [comment]
That's great! Thanks @borisstoyanov for your help!

**Comment by serg38** [comment]
LGTM
tag:mergeready

**Comment by rhtyd** [comment]
This needed to be merged on 4.9 as well since #1847 was merged on 4.9+ as well, this has now caused regressions in 4.9 branch. I'll have to backport the fixes on 4.9 now.

**Pull request merged**